packages/nimble-components/src/select/index.ts

```diff
@@ -837,6 +931,41 @@ export class Select
         this.committedSelectedOption = options[this.selectedIndex];
     }
 
+    private toggleNewActiveOption(newActiveIndex: number): void {
+        const selectedIndex = this.openActiveIndex ?? this.selectedIndex;
+        const activeOption = this.options[newActiveIndex];
```

Comment:
Should `activeOption = this.options[newActiveIndex]` move into the if-statement? Doesn't seem like it makes sense to do that assignment when you haven't checked that the `newActiveIndex` isn't -1.

Reply:
The code is set up to handle the case where the result of indexing with -1 results in an undefined value. There's been further code cleanup, however, that hopefully makes this intent more clear.